cmd/process-agent/collector.go

```diff
@@ -258,6 +260,35 @@ func (l *Collector) messagesToResultsQueue(start time.Time, name string, message
 	updateProcContainerCount(messages)
 }
 
+// getRequestID generates a unique identifier (string representation of 64 bits integer) that is composed as follows:
+//	1. 22 bits of the seconds in the current month.
+//	2. 28 bits of hash of the hostname and process agent pid.
+// 	3. 14 bits of the current message in the batch being sent to the server.
+func (l *Collector) getRequestID(start time.Time, chunkIndex int) (string, error) {
```

Comment:
This function looks correct, but could probably do with some testing.

There are also a couple of low hanging fruits for performance though I can't imagine this will be a concern.
1. we can use bitwise operations instead of generating a binary string  and calling strconf.ParseUint
2. the host name/pid hash could be cached

Reply:
absolute accurate!
Ran benchmarks before and after the change
![image](https://user-images.githubusercontent.com/17148247/193293959-0bcb50ab-baba-4f16-8175-c7edc6997b20.png)